src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useEffect, useMemo } from "react";
+import { Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useAsyncFn } from "react-use";
+import firebase from "firebase/app";
+
+import { MAX_TABLE_CAPACITY, MIN_TABLE_CAPACITY } from "settings";
+
+import { Table } from "types/Table";
+
+import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
+
+import { InputField } from "components/atoms/InputField";
+
+import "./EditTableTitleModal.scss";
+
+export interface EditTableForm {
+  title: string;
+  subtitle?: string;
+  capacity: number;
+}
+
+export interface EditTableTitleModalProps {
+  isShown: boolean;
+  title: string;
+  subtitle?: string;
+  capacity: number;
+  venueName: string;
+  tables: Table[];
+  tableOfUser?: Table;
+  onHide: () => void;
+}
+
+export const EditTableTitleModal: React.FC<EditTableTitleModalProps> = ({
+  isShown,
+  venueName,
+  title,
+  tables,
+  tableOfUser,
+  subtitle,
+  capacity,
+  onHide,
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+
+  const formDefaultValues = useMemo(
+    () => ({
+      title,
+      subtitle,
+      capacity,
+    }),
+    [capacity, subtitle, title]
+  );
+
+  const {
+    register,
+    handleSubmit,
+    errors,
+    watch,
+    reset,
+  } = useForm<EditTableForm>({
+    mode: "onBlur",
+    reValidateMode: "onBlur",
+    defaultValues: formDefaultValues,
+  });
+
+  // Updates the default values whenever they change. This is required because they are cached after the first render.
+  useEffect(() => {
+    reset(formDefaultValues);
+  }, [formDefaultValues, reset, title]);
```

Comment:
Looking at the docs for this, I expect we probably want to be setting a bunch of the props on this as well, not just blindly resetting the fields:

![image](https://user-images.githubusercontent.com/753891/119298473-73e35680-bca0-11eb-92b9-af2763df673a.png)

In particular, it would be really annoying if I was halfway through typing something, then the state changed and wiped out what I was typing, with no way for me to recover it.

I haven't added any of the extra options mentioned above to this yet (assuming we need them):

```suggestion
  useEffect(() => {
    reset({ title, subtitle, capacity });
  }, [reset, title, subtitle, capacity]);
```

Reply:
@denisdimitrov What was the need for implementing this reset out of curiosity? How often are the props likely to change without unrendering/re-rendering the component to necessitate this?

I'm guessing this has something to do with it being a modal, and thus probably re-used for many things rather than unrendered/re-rendered like most 'normal' components?

Reply:
If we are in the same table and you update the capacity or something else, it should update the data in my modal as well. And since `useForm` caches only the first time, we need this.